### ISSUE TITLE
fix: Circular constraint dependency involving 'Icon' and 'Icon' in FluentIcon

### ISSUE THREAD

**Issue by HofmeisterAn** [microsoft/fluentui-blazor#2137]
# 🐛 Bug Report

Hi, I have been using FluentUI for a while now, and I appreciate your work! Unfortunately, I am encountering an issue when building a simple service in Linux environments.

As soon as I add the `FluentIcon` component to a page, the build fails on Linux machines:

> /workspaces/FooBarBaz/src/Core/Microsoft.NET.Sdk.Razor.SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator/Pages_Index_razor.g.cs(291,47): error CS0454: Circular constraint dependency involving 'Icon' and 'Icon' [/workspaces/FooBarBaz/src/Core/Core.csproj]

Looking into the generated sources, the error makes sense. On **Windows** (where the build succeeds), the `Pages_Index_razor.g.cs` looks like this:

```csharp
public static void CreateFluentIcon_2<Icon>(global::Microsoft.AspNetCore.Components.Rendering.RenderTreeBuilder __builder, int seq, int __seq0, Icon __arg0, int __seq1, global::Microsoft.Fast.Components.FluentUI.Color? __arg1)
    where Icon : global::Microsoft.Fast.Components.FluentUI.Icon, new()
{
    __builder.OpenComponent<global::Microsoft.Fast.Components.FluentUI.FluentIcon<Icon>>(seq);
    __builder.AddAttribute(__seq0, "Value", (object)__arg0);
    __builder.AddAttribute(__seq1, "Color", (object)__arg1);
    __builder.CloseComponent();
}
```

However, on **Linux** where it fails, it looks like this (notice the type constraint [CS0454](https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0454)):

```csharp
public static void CreateFluentIcon_2<Icon>(global::Microsoft.AspNetCore.Components.Rendering.RenderTreeBuilder __builder, int seq, int __seq0, Icon __arg0, int __seq1, global::Microsoft.Fast.Components.FluentUI.Color? __arg1)
    where Icon : Icon, new()
{
    __builder.OpenComponent<global::Microsoft.Fast.Components.FluentUI.FluentIcon<Icon>>(seq);
    __builder.AddAttribute(__seq0, "Value", __arg0);
    __builder.AddAttribute(__seq1, "Color", __arg1);
    __builder.CloseComponent();
}
```

I cannot imagine many other developers are encountering this issue, as I haven't found any related reports. Right now, I have no idea why the generated sources differ and where the error comes from. Does anyone have any ideas?

## 💻 Repro or Code Sample

```razor
@page "/"

<FluentIcon Value="FilledCircle" Color="@Color.Accent"/>

@code
{
    private static readonly Icon FilledCircle = new Icons.Filled.Size16.Circle();
}
```

## 🤔 Expected Behavior

- Builds on Linux environments do not encounter CS0454 and succeed.

## 😯 Current Behavior

- Builds on Linux environments encounter CS0454 and fail.

## 💁 Possible Solution

\-

## 🔦 Context

\-

## 🌍 Your Environment

* OS & Device: `mcr.microsoft.com/devcontainers/base:jammy`, `ubuntu-22.04`
* .NET and Fluent UI Blazor library Version:
    * `.NET@6.0.411`
    * `Microsoft.Fast.Components.FluentUI@3.7.1`
    * `Microsoft.Fast.Components.FluentUI.Icons@3.7.1`

**Comment by vnbaaij** [comment]
Hi,
I have no idea why the generated code would be different. Probably not found before because not many are building directly on Linux?

Not sure if it would have any effect, but any specific reason why you are running on .NET 6. I know a lot of investments have been made to make .NET 8 run better on Linux. Also, we focus on our .NET 8 version (Microsoft.FluentUI.AspNetCore.Components) where our .NET 6/7 version is in 'maintenance mode'.

If you are 'stuck' on .NET 6, I would advice you to open an issue in the dotnet repo as I think this is more a low level thing than something we can solve on our side.

**Comment by HofmeisterAn** [comment]
> but any specific reason why you are running on .NET 6.

This is a kind of legacy project. We are planning to migrate it by the end of .NET 6 EOL (which is soon), along with some other projects. Anyway, I tried the same page with .NET 8 (8.0.205) and `Microsoft.FluentUI.AspNetCore.Components` version `4.7.2` and can confirm that the build is working ✔️.

> If you are 'stuck' on .NET 6, I would advice you to open an issue in the dotnet repo as I think this is more a low level thing than something we can solve on our side.

I double-checked the generated sources. Using .NET 8 uses the full namespace as well (like the Windows build). It looks like I ran into a bug in the .NET 6 version (SDK/source generator).

For anyone who encounters this issue using .NET 6, it looks like explicitly specifying the generic type resolves/workarounds it:

    <FluentIcon Icon="Microsoft.Fast.Components.FluentUI.Icon" Value="FilledCircle" Color="@Color.Accent"/>

Thanks for you reply and work.
